Framework/MDAlgorithms/src/SaveMD2.cpp

```diff
@@ -104,28 +116,36 @@ void SaveMD2::doSaveHisto(Mantid::DataObjects::MDHistoWorkspace_sptr ws) {
   file->putAttr("QConvention", m_QConvention);
 
   // Write out the visual normalization
-  file->writeData("visual_normalization",
-                  static_cast<uint32_t>(ws->displayNormalization()));
+  if (getProperty("SaveSample"))
+    file->writeData("visual_normalization",
+                    static_cast<uint32_t>(ws->displayNormalization()));
 
   // Save the algorithm history under "process"
-  ws->getHistory().saveNexus(file);
+  if (getProperty("SaveHistory"))
+    ws->getHistory().saveNexus(file);
```

Comment:
Just a note, this will give a warning on load `Error opening the algorithm history field 'process'. Workspace will have no history.`